cmake/Subdirectories.cmake

```
@@ -202,7 +203,7 @@ function(resolve_gem_dependencies object_type object_path)
     set(ENV{PYTHONNOUSERSITE} 1)
     string(TOLOWER ${object_type} object_type_lower)
     execute_process(COMMAND 
-        ${LY_PYTHON_CMD} "${LY_ROOT_FOLDER}/scripts/o3de/o3de/cmake.py" --${object_type_lower}-path "${object_path}"
+        ${LY_PYTHON_CMD} "${LY_ROOT_FOLDER}/scripts/o3de/o3de/cmake.py" --${object_type_lower}-path "${object_path}" -ed "${O3DE_EXTERNAL_SUBDIRS}"
```

To make sure we are retrieving O3DE_EXTERNAL_SUBDIRS from the cache and not some override local variable we can use `get_property(<var name> CACHE PROPERTY O3DE_EXTERNAL_SUBDIRS)`

We also should only specify the `-ed` option if the list of user provided subdirectories list is not empty

```suggestion
        get_property(user_external_subdirs CACHE PROPERTY O3DE_EXTERNAL_SUBDIRS)
        if(user_external_subdirs)
            set(user_external_subdir_option -ed "${user_external_subdirs}")
        endif()
        ${LY_PYTHON_CMD} "${LY_ROOT_FOLDER}/scripts/o3de/o3de/cmake.py" --${object_type_lower}-path "${object_path}" ${user_external_subdir_option}
```